internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -571,11 +581,49 @@ func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m
 	return dataSourceTMCEKSClusterRead(ctx, d, m)
 }
 
+func resourceClusterImporter(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	config := m.(authctx.TanzuContext)
+
+	id := d.Id()
+	if id == "" {
+		return nil, errors.New("ID is needed to import an TMC EKS cluster")
+	}
+
+	resp, err := config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceGetByID(id)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS cluster entry for id %s", id)
+	}
+
+	npresp, err := config.TMCConnection.EKSNodePoolResourceService.EksNodePoolResourceServiceList(resp.EksCluster.FullName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS nodepools for cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(CredentialNameKey, resp.EksCluster.FullName.CredentialName); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set credential name for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(RegionKey, resp.EksCluster.FullName.Region); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set region for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	if err = d.Set(NameKey, resp.EksCluster.FullName.Name); err != nil {
+		return nil, errors.Wrapf(err, "Failed to set name for the cluster %s", resp.EksCluster.FullName.Name)
+	}
+
+	err = setResourceData(d, resp.EksCluster, npresp.Nodepools)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to set resourcee data during import for %s", resp.EksCluster.FullName.Name)
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
 func handleClusterDiff(config authctx.TanzuContext, tmcCluster *eksmodel.VmwareTanzuManageV1alpha1EksclusterEksCluster, meta *objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta, clusterSpec *eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec) error {
 	updateCluster := false
 
 	if meta.Description != tmcCluster.Meta.Description ||
-		!reflect.DeepEqual(meta.Labels, tmcCluster.Meta.Labels) {
+		!mapEqual(meta.Labels, tmcCluster.Meta.Labels) {
```

Comment:
nit question: why is this change needed?

Reply:
When the labels field is empty in one and not set in the other, `reflect.DeepEqual` leads to a false positive of change.